doc/modules/mod_mam.md

```diff
@@ -145,6 +145,61 @@ You can change the default settings using extra parameters:
 }.
 ```
 
+### Custom backend
+
+You can implement a custom backend for mod_mam. In this case the module needs to implement the `gen_mod` behavior and make it have its own toplevel place within the modules section of your configuration file, eg:
+
+``` erlang
+{modules, [
+  ...
+  {mod_mam_custom_arch, []},
+  {mod_mam, [...]},
+  ...
+]}
+```
+
+Typically your module would add handlers for hooks triggered by mod_mam. See other backend implementations to get an idea.
+
+Eg.:
+
+``` erlang
+start(Host, _Opts) ->
+  ...
+  ejabberd_hooks:add(mam_lookup_messages, Host, ?MODULE, lookup_messages, 50),
+  ...
```

Comment:
This description should be more precise and I'd actually move it to a new page in "Developer's Guide" and leave a reference here.

If you don't have time to create a better description, I'd appreciate at least if you could move it to a new page.